File: backend/models/advisor.py
from pydantic import BaseModel
from typing import List, Optional


class AdvisorRequest(BaseModel):
    query: str


class RecommendedProduct(BaseModel):
    name: str
    price: float
    reason: str
    confidence: int
    trendyol_url: str = ""
    akakce_url: str = ""
    google_shopping_url: str = ""
    trend_direction: str = "STABIL"
    trend_score: int = 0
    youtube_video_count: int = 0
    youtube_latest_url: str = ""
    shopgrill_price: Optional[float] = None
    carsila_price: Optional[float] = None
    shopgrill_url: Optional[str] = None
    carsila_url: Optional[str] = None
    cheaper_store: Optional[str] = None
    store_note: Optional[str] = None


class AdvisorResponse(BaseModel):
    recommendations: List[RecommendedProduct]

File: frontend/lib/advisor.ts
import { apiFetch } from "./api";

export type RecommendedProduct = {
  name: string;
  price: number;
  reason: string;
  confidence: number;
  trendyol_url?: string;
  akakce_url?: string;
  google_shopping_url?: string;
  trend_direction?: string;
  trend_score?: number;
  youtube_video_count?: number;
  youtube_latest_url?: string;
};

export type AdvisorResponse = {
  recommendations: RecommendedProduct[];
};

export function askSmartAdvisor(query: string) {
  return apiFetch<AdvisorResponse>("/smart-advisor", {
    method: "POST",
    body: JSON.stringify({ query }),
  });
}
